pkg/utils/disruption/disruption.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disruption
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/clock"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+// lifetimeRemaining calculates the fraction of node lifetime remaining in the range [0.0, 1.0].  If the TTLSecondsUntilExpired
+// is non-zero, we use it to scale down the disruption costs of candidates that are going to expire.  Just after creation, the
+// disruption cost is highest, and it approaches zero as the node ages towards its expiration time.
+func LifetimeRemaining(clock clock.Clock, nodePool *v1beta1.NodePool, node *v1.Node) float64 {
+	remaining := 1.0
+	if nodePool.Spec.Disruption.ExpireAfter.Duration != nil {
+		ageInSeconds := clock.Since(node.CreationTimestamp.Time).Seconds()
+		totalLifetimeSeconds := nodePool.Spec.Disruption.ExpireAfter.Duration.Seconds()
+		lifetimeRemainingSeconds := totalLifetimeSeconds - ageInSeconds
+		remaining = lo.Clamp(lifetimeRemainingSeconds/totalLifetimeSeconds, 0.0, 1.0)
+	}
+	return remaining
+}
+
+// GetPodEvictionCost returns the disruption cost computed for evicting the given pod.
+func GetPodEvictionCost(ctx context.Context, p *v1.Pod) float64 {
```

Comment:
Wdyt about dropping `Get` and `Pod` like
```suggestion
func EvictionCost(ctx context.Context, p *v1.Pod) float64 {
```

Reply:
Good call!